wallet/wallet.c

```diff
@@ -661,8 +661,9 @@ static bool wallet_ever_used(struct wallet *w)
 	/* Or if they do a unilateral close, the output to us provides a UTXO. */
 	stmt = db_query(__func__, w->db,
 			"SELECT COUNT(*) FROM outputs WHERE commitment_point IS NOT NULL;");
-	sqlite3_step(stmt);
-	channel_utxos = (sqlite3_column_int(stmt, 0) != 0);
+	if (sqlite3_step(stmt) == SQLITE_ROW) {
```

Comment:
For this case, `SELECT COUNT(*)` will always assuredly return a single row.

I would have written instead as an assert:

    ret = sqlite3_step(stmt);
    assert(ret == SQLITE_ROW);
    channel_utxos = (sqlite3_column_int(stmt, 0) != 0);
    ret = sqlite3_step(stmt);
    assert(ret == SQLITE_DONE);
    (void) ret;

The above expresses the assertions that we expect a single row and that after that single row, there are no more rows.